tools/clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5097,14 +5097,65 @@ static void ProcessDeclAttribute(Sema &S, Scope *scope, Decl *D,
   }
 }
 
+void Sema::ValidateShaderAttribtues(Scope *S, Decl *D, const AttributeList *A) {
```

Comment:
I don't think this is the right place to do this validation. We should probably do it off `HandleDeclAttributeForHLSL`